lib/hls/hls_parser.js

```diff
@@ -1402,25 +1403,72 @@ shaka.hls.HlsParser = class {
     }
 
     let encrypted = false;
+    let clearKey;
     /** @type {!Array.<shaka.extern.DrmInfo>}*/
     const drmInfos = [];
     const keyIds = new Set();
 
-    // TODO: May still need changes to support key rotation.
+    /* eslint-disable no-await-in-loop */
+    // TODO: May still need changes to support key rotation and discontinuity.
     for (const drmTag of drmTags) {
       const method = drmTag.getRequiredAttrValue('METHOD');
-      if (method != 'NONE') {
+      if (method == 'AES-128') {
         encrypted = true;
+        this.aesEncrypted_ = true;
 
-        // We do not support AES-128 encryption with HLS yet. So, do not create
-        // StreamInfo for the playlist encrypted with AES-128.
-        // TODO: Remove the error message once we add support for AES-128.
-        if (method == 'AES-128') {
-          shaka.log.warning('Unsupported HLS Encryption', method);
-          this.aesEncrypted_ = true;
-          return null;
+        // Check if the Web Crypto API is available.
+        if (!window.crypto || !window.crypto.subtle) {
+          shaka.log.alwaysWarn('Web Crypto API is not available to decrypt ' +
+              'AES-128. (Web Crypto only exists in secure origins like https)');
+          throw new shaka.util.Error(
+              shaka.util.Error.Severity.CRITICAL,
+              shaka.util.Error.Category.MANIFEST,
+              shaka.util.Error.Code.HLS_KEYFORMATS_NOT_SUPPORTED);
```

Comment:
I think it is better to add a new error code for this case.

Reply:
Thanks. Done.